riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.java

```diff
@@ -0,0 +1,264 @@
+
+        private final Collection<Endpoint<?>> endpointsToValidate;
+
+        public TestRequestSecurityValidator(Collection<Endpoint<?>> endpointsToValidate) {
+            this.endpointsToValidate = endpointsToValidate;
+        }
+
+        @Override
+        public void validateSecureRequestForEndpoint(RequestInfo<?> requestInfo, Endpoint<?> endpoint) {
+            if ("true".equals(requestInfo.getHeaders().get("force-security-error"))) {
+                throw new Unauthorized401Exception("Forcing Security Error.", requestInfo.getPath(), null);
+            }
+        }
+
+        @Override
+        public Collection<Endpoint<?>> endpointsToValidate() {
+            return endpointsToValidate;
+        }
+    }
+
+    private static class BasicEndpoint extends StandardEndpoint<Void, String> {
+
+        public static final String MATCHING_PATH = "/basicEndpoint";
+        public static final String RESPONSE_PAYLOAD = "basic-endpoint-" + UUID.randomUUID().toString();
+        public static final String FORCE_ERROR_HEADER_KEY = "force-error";
+        public static final ApiError FORCED_ERROR = new ApiErrorBase("FORCED_ERROR", 42, "forced error", 542);
+
+        @Override
+        public CompletableFuture<ResponseInfo<String>> execute(RequestInfo<Void> request, Executor longRunningTaskExecutor, ChannelHandlerContext ctx) {
+            if ("true".equals(request.getHeaders().get(FORCE_ERROR_HEADER_KEY)))
+                throw ApiException.newBuilder().withApiErrors(FORCED_ERROR).build();
+
+            return CompletableFuture.completedFuture(
+                    ResponseInfo.newBuilder(RESPONSE_PAYLOAD).build()
+            );
+        }
+
+        @Override
+        public Matcher requestMatcher() {
+            return Matcher.match(MATCHING_PATH, HttpMethod.GET);
+        }
+
+    }
+
+    protected static class BeforeSecurityRequestFilter implements RequestAndResponseFilter {
+
+        public static final String BEFORE_SECURITY_FILTER_FIRST_CHUNK_REQ_HEADER_KEY = "req-header-BEFORE-SECURITY-filter-first-chunk";
+        public static final String BEFORE_SECURITY_FILTER_FIRST_CHUNK_REQ_HEADER_VALUE = UUID.randomUUID().toString();
+
+        public static final String BEFORE_SECURITY_FILTER_LAST_CHUNK_REQ_HEADER_KEY = "req-header-BEFORE-SECURITY-filter-last-chunk";
+        public static final String BEFORE_SECURITY_FILTER_LAST_CHUNK_REQ_HEADER_VALUE = UUID.randomUUID().toString();
+
+        public static final String BEFORE_SECURITY_FILTER_RESPONSE_HEADER_KEY = "response-header-BEFORE-SECURITY-filter";
+        public static final String BEFORE_SECURITY_FILTER_RESPONSE_HEADER_VALUE = UUID.randomUUID().toString();
+
+        @Override
+        public <T> RequestInfo<T> filterRequestFirstChunkNoPayload(RequestInfo<T> currentRequestInfo, ChannelHandlerContext ctx) {
+            currentRequestInfo.getHeaders().set(BEFORE_SECURITY_FILTER_FIRST_CHUNK_REQ_HEADER_KEY, BEFORE_SECURITY_FILTER_FIRST_CHUNK_REQ_HEADER_VALUE);
```

Comment:
Have the filters set attributes on the request, not headers. Attributes are serverside-only and thus not affected by headers sent by the caller.